django/db/backends/base/schema.py

```diff
@@ -1708,19 +1695,28 @@ def _field_should_be_altered(self, old_field, new_field, ignore=None):
         ):
             old_kwargs.pop("to", None)
             new_kwargs.pop("to", None)
-        # db_default can take many form but result in the same SQL.
+        # db_default can take many forms but result in the same SQL.
         if (
             old_kwargs.get("db_default")
             and new_kwargs.get("db_default")
             and self.db_default_sql(old_field) == self.db_default_sql(new_field)
         ):
             old_kwargs.pop("db_default")
             new_kwargs.pop("db_default")
-        return (
-            old_field.concrete
-            and new_field.concrete
-            and (self.quote_name(old_field.column) != self.quote_name(new_field.column))
-        ) or (old_path, old_args, old_kwargs) != (new_path, new_args, new_kwargs)
+
+        old_schema_name = (
+            self.quote_name(old_field.column)
+            if not old_field.many_to_many
+            else old_field.attname
+        )
+        new_schema_name = (
+            self.quote_name(new_field.column)
+            if not new_field.many_to_many
+            else new_field.attname
+        )
```

Comment:
@cliffordgama After having a look at #20412, do you have an opinion about how to factor this? I think the multiple returns helps readability.